library/net.c

```diff
@@ -944,11 +944,16 @@ int dnet_state_micro_init(struct dnet_net_state *st,
 	st->n = n;
 
 	st->la = 1;
-	st->weight = DNET_STATE_DEFAULT_WEIGHT;
 
 	INIT_LIST_HEAD(&st->node_entry);
 	INIT_LIST_HEAD(&st->storage_state_entry);
-	INIT_LIST_HEAD(&st->idc_list);
+	st->idc_root = RB_ROOT;
+	err = pthread_rwlock_init(&st->idc_lock, NULL);
```

Comment:
`st->id_lock` will not be destroyed if `pthread_mutex_init(&st->trans_lock, NULL)` is failed